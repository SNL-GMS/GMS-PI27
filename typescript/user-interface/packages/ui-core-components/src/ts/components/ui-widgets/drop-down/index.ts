export type { DropDownProps } from './drop-down';
export { DropDown } from './drop-down';
