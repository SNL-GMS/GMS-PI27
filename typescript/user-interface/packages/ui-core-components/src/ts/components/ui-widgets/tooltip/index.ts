import { HotkeyTooltip } from './hotkey-tooltip';
import { TooltipWrapper } from './tooltip';

export { HotkeyTooltip, TooltipWrapper as Tooltip2Wrapper };
