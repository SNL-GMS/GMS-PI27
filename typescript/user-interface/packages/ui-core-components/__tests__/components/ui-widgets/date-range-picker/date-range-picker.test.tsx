/* eslint-disable @typescript-eslint/no-magic-numbers */
import {
  DATE_TIME_FORMAT_WITH_FRACTIONAL_SECOND_PRECISION,
  DATE_TIME_FORMAT_WITH_SECOND_PRECISION
} from '@gms/common-util';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';

import { DateRangePicker } from '../../../../src/ts/components/ui-widgets/date-range-picker';

const MOCK_TIME = 1609506000000;

const lodash = jest.requireActual('lodash');
lodash.uniqueId = () => '1';

describe('DateRangePicker', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(MOCK_TIME);
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should be defined', () => {
    expect(DateRangePicker).toBeDefined();
  });

  it('matches the snapshot', () => {
    const { container } = render(
      <DateRangePicker
        durations={[
          {
            description: 'one second',
            value: 1000
          },
          {
            description: 'two seconds',
            value: 2000
          }
        ]}
        startTimeMs={MOCK_TIME - 2000}
        endTimeMs={MOCK_TIME}
        onNewInterval={jest.fn()}
        onApply={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot with fractional second precision format', () => {
    const { container } = render(
      <DateRangePicker
        durations={[
          {
            description: 'one second',
            value: 1000
          },
          {
            description: 'two seconds',
            value: 2000
          }
        ]}
        startTimeMs={MOCK_TIME - 2000}
        endTimeMs={MOCK_TIME}
        onNewInterval={jest.fn()}
        onApply={jest.fn()}
        format={DATE_TIME_FORMAT_WITH_FRACTIONAL_SECOND_PRECISION}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('matches the snapshot with second precision format', () => {
    const { container } = render(
      <DateRangePicker
        durations={[
          {
            description: 'one second',
            value: 1000
          },
          {
            description: 'two seconds',
            value: 2000
          }
        ]}
        startTimeMs={MOCK_TIME - 2000}
        endTimeMs={MOCK_TIME}
        onNewInterval={jest.fn()}
        onApply={jest.fn()}
        format={DATE_TIME_FORMAT_WITH_SECOND_PRECISION}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('opens and closes the popup', async () => {
    render(
      <DateRangePicker
        durations={[
          {
            description: 'one second',
            value: 1000
          },
          {
            description: 'two seconds',
            value: 2000
          }
        ]}
        startTimeMs={MOCK_TIME - 2000}
        endTimeMs={MOCK_TIME}
        onNewInterval={jest.fn()}
        onApply={jest.fn()}
      />
    );

    fireEvent.click(screen.getByRole('button'));

    expect(screen.queryByRole('dialog')).toBeDefined();

    fireEvent.click(screen.getByText('Cancel'));

    await waitFor(() => {
      expect(screen.queryByRole('dialog')).toBeNull();
    });
  });
});
