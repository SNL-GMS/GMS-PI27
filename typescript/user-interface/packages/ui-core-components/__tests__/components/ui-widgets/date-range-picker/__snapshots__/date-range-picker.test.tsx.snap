// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker matches the snapshot 1`] = `
<div>
  <div
    class="date-range-picker"
    id="date-range-picker-id-1"
  >
    <label
      class="bp5-label bp5-label bp5-inline date-range-picker__label"
    >
      Time Range: 2021-01-01 12:59 to 2021-01-01 13:00
    </label>
    <div>
      <button
        class="bp5-button date-range-picker__apply-button"
        data-cy="date-range-picker-edit"
        type="button"
      >
        <span
          class="bp5-button-text"
        >
          Edit
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`DateRangePicker matches the snapshot with fractional second precision format 1`] = `
<div>
  <div
    class="date-range-picker"
    id="date-range-picker-id-2"
  >
    <label
      class="bp5-label bp5-label bp5-inline date-range-picker__label"
    >
      Time Range: 2021-01-01 12:59:58.000 to 2021-01-01 13:00:00.000
    </label>
    <div>
      <button
        class="bp5-button date-range-picker__apply-button"
        data-cy="date-range-picker-edit"
        type="button"
      >
        <span
          class="bp5-button-text"
        >
          Edit
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`DateRangePicker matches the snapshot with second precision format 1`] = `
<div>
  <div
    class="date-range-picker"
    id="date-range-picker-id-3"
  >
    <label
      class="bp5-label bp5-label bp5-inline date-range-picker__label"
    >
      Time Range: 2021-01-01 12:59:58 to 2021-01-01 13:00:00
    </label>
    <div>
      <button
        class="bp5-button date-range-picker__apply-button"
        data-cy="date-range-picker-edit"
        type="button"
      >
        <span
          class="bp5-button-text"
        >
          Edit
        </span>
      </button>
    </div>
  </div>
</div>
`;
